usr/src/uts/i86pc/io/vmm/vmm_sol_dev.c

```diff
@@ -1205,26 +1206,37 @@ vmmdev_do_vm_create(char *name, cred_t *cr)
 	vmm_softc_t	*sc = NULL;
 	minor_t		minor;
 	int		error = ENOMEM;
+	hma_reg_t	*reg = NULL;
 
 	if (strnlen(name, VM_MAX_NAMELEN) >= VM_MAX_NAMELEN) {
 		return (EINVAL);
 	}
 
+	mutex_enter(&vmmdev_mtx);
```

Comment:
this is basically reverting a good portion of:

14f894282dbb1af80efba29ec886e852b04ca13c OS-7446 clean up bhyve locking and names

Any reason we can't register-then-drop the mutex, so we don't need to hold it across the whole operation?